packages/web/app/src/pages/target-laboratory.tsx

```diff
@@ -251,17 +253,17 @@ function Save(props: {
   );
 }
 
-function substituteVariablesInHeader(
+function substituteVariablesInHeaders(
   headers: Record<string, string>,
-  environmentVariables: Record<string, unknown>,
+  environmentVariables: Record<string, Kit.JSON.Value>,
 ) {
   return Object.fromEntries(
     Object.entries(headers).map(([key, value]) => {
       if (typeof value === 'string') {
         // Replace all occurrences of `{{keyName}}` strings only if key exists in `environmentVariables`
-        value = value.replaceAll(/{{(?<keyName>.*?)}}/g, (originalString, envKey) => {
+        value = value.replaceAll(/{{(?<keyName>.*?)}}/g, (originalString, envKey: string) => {
           return Object.hasOwn(environmentVariables, envKey)
-            ? (environmentVariables[envKey] as string)
+            ? String(environmentVariables[envKey])
             : originalString;
         });
       }
```

Comment:
Having specified the index type of environment variables beyond `unknown` upstream, it led me here. This code now explicitly handles non-strings.